### PR TITLE
docker_image_history table

### PR DESCRIPTION
**Pull request by spasam**
This table provides output similar to "docker history" command.
See: https://docs.docker.com/engine/reference/commandline/history/
created_by column has useful information related to the command history

Sample output:
```
osquery> select * from docker_image_history where id = '9499db781771';
+--------------+------------+-----------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+---------+
| id           | created    | size      | created_by                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | tags         | comment |
+--------------+------------+-----------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+---------+
| 9499db781771 | 1606343174 | 0         | /bin/sh -c #(nop)  CMD ["/bin/bash"]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | ubuntu:16.04 |         |
| 9499db781771 | 1606343174 | 7         | /bin/sh -c mkdir -p /run/systemd && echo 'docker' > /run/systemd/container                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |              |         |
| 9499db781771 | 1606343173 | 0         | /bin/sh -c rm -rf /var/lib/apt/lists/*                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |              |         |
| 9499db781771 | 1606343172 | 745       | /bin/sh -c set -xe 		&& echo '#!/bin/sh' > /usr/sbin/policy-rc.d 	&& echo 'exit 101' >> /usr/sbin/policy-rc.d 	&& chmod +x /usr/sbin/policy-rc.d 		&& dpkg-divert --local --rename --add /sbin/initctl 	&& cp -a /usr/sbin/policy-rc.d /sbin/initctl 	&& sed -i 's/^exit.*/exit 0/' /sbin/initctl 		&& echo 'force-unsafe-io' > /etc/dpkg/dpkg.cfg.d/docker-apt-speedup 		&& echo 'DPkg::Post-Invoke { "rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true"; };' > /etc/apt/apt.conf.d/docker-clean 	&& echo 'APT::Update::Post-Invoke { "rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true"; };' >> /etc/apt/apt.conf.d/docker-clean 	&& echo 'Dir::Cache::pkgcache ""; Dir::Cache::srcpkgcache "";' >> /etc/apt/apt.conf.d/docker-clean 		&& echo 'Acquire::Languages "none";' > /etc/apt/apt.conf.d/docker-no-languages 		&& echo 'Acquire::GzipIndexes "true"; Acquire::CompressionTypes::Order:: "gz";' > /etc/apt/apt.conf.d/docker-gzip-indexes 		&& echo 'Apt::AutoRemove::SuggestsImportant "false";' > /etc/apt/apt.conf.d/docker-autoremove-suggests |              |         |
| 9499db781771 | 1606343171 | 130659802 | /bin/sh -c #(nop) ADD file:8eef54430e581236e6d529a7d09df648f43c840e889d9ae132e5ed25d7bd2b88 in /                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |              |         |
+--------------+------------+-----------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+---------+
```

### CODE REVIEW

File: osquery/tables/applications/posix/docker.cpp [hunk 1]
```diff
@@ -1096,6 +1096,84 @@ QueryData genImageLayers(QueryContext& context) {
   return results;
 }
 
+/**
+ * @brief Image history extractor for docker_image_history table
+ */
+void getImageHistory(std::string image_id, QueryData& results) {
```
Comment (theopolis) [inline osquery/tables/applications/posix/docker.cpp:1102]:
```suggestion
void getImageHistory(const std::string& image_id, QueryData& results) {
```

**Review by theopolis — CHANGES_REQUESTED** [review]
Looks good, one quick recommendation

File: osquery/tables/applications/posix/docker.cpp (hunk 1 above)
Comment (spasam) [reply osquery/tables/applications/posix/docker.cpp]: Thanks @theopolis . Fixed this and another function with similar signature.

File: osquery/tables/applications/posix/docker.cpp [hunk 2]
```diff
@@ -1018,7 +1018,7 @@ QueryData genVolumeLabels(QueryContext& context) {
 /**
  * @brief Image layer extractor for docker_image_layers table
  */
-void getImageLayers(std::string image_id, QueryData& results) {
+void getImageLayers(const std::string image_id, QueryData& results) {
```
Comment (theopolis) [inline osquery/tables/applications/posix/docker.cpp:1021]:
```suggestion
void getImageLayers(const std::string& image_id, QueryData& results) {
```

File: osquery/tables/applications/posix/docker.cpp [hunk 3]
```diff
@@ -1096,6 +1096,84 @@ QueryData genImageLayers(QueryContext& context) {
   return results;
 }
 
+/**
+ * @brief Image history extractor for docker_image_history table
+ */
+void getImageHistory(const std::string image_id, QueryData& results) {
```
Comment (theopolis) [inline osquery/tables/applications/posix/docker.cpp:1102]:
```suggestion
void getImageHistory(const std::string& image_id, QueryData& results) {
```

**Review by theopolis — CHANGES_REQUESTED** [review]

File: osquery/tables/applications/posix/docker.cpp (hunk 2 above)
Comment (spasam) [reply osquery/tables/applications/posix/docker.cpp:1021]: Crap, missed that from you original comment @theopolis . Sorry. Fixing

**Review by theopolis — APPROVED** [review]

**Pull request merged**